packages/iTwinUI-react/src/core/utils/components/MiddleTextTruncation.tsx

```diff
@@ -19,33 +19,37 @@ export type MiddleTextTruncationProps = {
    */
   endCharsCount?: number;
   /**
-   * Callback fired when text is truncated.
+   * Custom renderer for the truncated text.
    */
-  onTruncate?: (isTruncated: boolean) => void;
+  textRenderer?: (
+    truncatedText: string,
+    originalText: string,
+  ) => React.ReactNode;
 } & CommonProps;
-
 /**
  * Truncates text with the ellipsis in the middle,
  * leaving defined number of chars at the end.
  * @example
  * <MiddleTextTruncation text='ThisIsMyVeryLongFileName.dgn' />
+ * @example
+ * <MiddleTextTruncation text='ThisIsMyVeryLongFileName.dgn' endCharsCount={10} />
+ * @example
+ * <MiddleTextTruncation text='ThisIsMyVeryLongFileName.dgn' textRenderer={(truncatedText, originalText) => <span title={originalText}>{truncatedText}</span>} />
```

Comment:
Good that you added examples, but I think we want to recommend memoizing the `textRenderer`, so use `useCallback` in here and in stories too, to avoid not needed rerenders.

Reply:
Added callbacks